syntax = "proto3";

service Simulator {
    rpc RegisterEventReceiver(RegisterEventReceiverRequest) returns (RegisterEventReceiverResponse) {}
    rpc GetTimeStamp(GetTimeStampRequest) returns (GetTimeStampResponse) {}
    rpc GetNewCommandId(GetNewCommandIdRequest) returns (GetNewCommandIdResponse) {}
}

message RegisterEventReceiverRequest{
    string id = 1;
    string ip_addr = 2;
    int32 port = 3;
}

message RegisterEventReceiverResponse{
    bool success = 1;
}

message GetTimeStampRequest{
}

message GetTimeStampResponse{
    string timestamp = 1;
}

message GetNewCommandIdRequest{
}

message GetNewCommandIdResponse{
    int32 command_id = 1;
}